src/Assets/Asset.php

```diff
@@ -1692,27 +1700,142 @@ public function set_as_registered() {
 		return $this;
 	}
 
+	/**
+	 * Gives us the option to call any wp_*_script or wp_*_style function
+	 * on the asset.
+	 *
+	 * example: $asset->register_asset( ...$args ); will call wp_register_script( $slug, ...$args ); or
+	 * wp_register_style( $slug, ...$args ); based on if the asset is JS or CSS type.
+	 *
+	 * @since TBD
+	 *
+	 * @return static
+	 */
+	public function __call( $method, $args ) {
```

Comment:
This is smart and saves some lines of code, but I do not think makes for readable code.
I would unroll the `__call` method and explicitly create each method that is now part of the `@method` annotation.
This should likely not be a gateway to **all** `wp_` functions, but to a limited, controlled and **tested** set of them.
What would this do if I call `$asset->footer`?
If you end up having to maintain a safe-list of functions that can be called, you're better served by an explicit API made of concrete methods.